Sources/PackageCollections/PackageCollections.swift

```diff
@@ -67,6 +67,14 @@ public struct PackageCollections: PackageCollectionsProtocol {
         }
     }
 
+    public mutating func updateAuthTokens(_ authTokens: [AuthTokenType: String]?) {
```

Comment:
Not my favorite approach. Open to other options. I don't think the API should be added to `PackageCollectionsProtocol` though.

Reply:
what about a setter?

Reply:
also another approach is to define Configuration::authToken to take a function that is called to retrieve the auth values


```
var authTokens = () -> [AuthTokenType: String]?
```

Reply:
> also another approach is to define Configuration::authToken to take a function

I like this. Thanks. Changed impl to this instead. https://github.com/apple/swift-package-manager/pull/3443/commits/bb67334894b5794c307ccf31cb7bb4ada7bcf015

Reply:
The approach in https://github.com/apple/swift-package-manager/pull/3443/commits/bb67334894b5794c307ccf31cb7bb4ada7bcf015 is a breaking change though, so my original approach might be safer... 🤔

Reply:
I think its okay to break at this point if we think its the right API, we just need to coordinate landing it cc @neonichu